File: c2-chaincode/chaincode-api/chaincode-api-fabric/src/main/java/io/komune/c2/chaincode/api/fabric/config/HasTlsCacerts.java
package io.komune.c2.chaincode.api.fabric.config;

import com.google.common.base.Strings;
import io.komune.c2.chaincode.api.fabric.utils.FileUtils;

import java.io.IOException;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.Properties;

public interface HasTlsCacerts {

    String getTlsCacerts();

    default URL getTlsCacertsAsUrl(String cryptoBase) throws MalformedURLException {
        return FileUtils.getUrl(cryptoBase, getTlsCacerts());
    }

    default Properties getPeerTlsProperties(String cryptoBase) throws IOException {
        Properties prop = new Properties();
        prop.setProperty("allowAllHostNames", "true");
        URL path = getTlsCacertsAsUrl(cryptoBase);
        prop.setProperty("pemFile", path.getFile());
        return prop;
    }
}

File: c2-chaincode/chaincode-api/chaincode-api-config/src/main/kotlin/io/komune/c2/chaincode/api/config/utils/HasTlsCacerts.kt
package io.komune.c2.chaincode.api.config.utils

import java.io.IOException
import java.net.MalformedURLException
import java.net.URL
import java.util.Properties


interface HasTlsCacerts {
    val tlsCacerts: String

    @Throws(MalformedURLException::class)
    fun getTlsCacertsAsUrl(cryptoBase: String): URL {
        return FileUtils.getUrl(cryptoBase, tlsCacerts)
    }

    @Throws(IOException::class)
    fun getPeerTlsProperties(cryptoBase: String): Properties {
        val prop = Properties()
        prop.setProperty("allowAllHostNames", "true")
        val path = getTlsCacertsAsUrl(cryptoBase)
        prop.setProperty("pemFile", path.file)
        return prop
    }
}
